_episodes/07-github.md

```diff
@@ -112,22 +119,188 @@ $ git remote -v
+origin   git@github.com:vlad/planets.git (push)
 ~~~
 {: .output}
 
 We'll discuss remotes in more detail in the next episode, while
 talking about how they might be used for collaboration.
 
-Once the remote is set up, this command will push the changes from
+## SSH Background and Setup
+Before Dracula can connect to a remote repository, he needs to set up a way for his computer to authenticate with GitHub so it knows it’s him trying to connect to his remote repository. 
+
+We are going to set up the method that is commonly used by many different services to authenticate access on the command line.  This method is SSH, or secure shell protocol.  SSH is a cryptographic network protocol that allows secure communication between computers using an otherwise insecure network.  
+
+SSH uses what is called a key pair. This is two keys that work together to validate access. One key is publicly known and called the public key, and the other key called the private key is kept private. Very descriptive names.
+
+You can think of the public key as a padlock, and only you have the key (the private key) to open it. You use the public key where you want a secure method of communication, such as your GitHub account.  You give this padlock, or public key, to GitHub and say “lock the communications to my account with this so that only computers that have my private key can unlock communications and send git commands as my GitHub account.”  
+
+What we will do now is do the minimum required to set up the SSH keys and add the public key to a GitHub account.
+
+> ## Advanced SSH
+> A supplemental episode in this lesson discusses SSH and key pairs in more depth and detail. 
+{: .callout}
+
+The first thing we are going to do is check if this has already been done on the computer you’re on.  Because generally speaking, this setup only needs to happen once and then you can forget about it. 
+
+> ## Keeping your keys secure
+> You shouldn't really forget about your SSH keys, since they keep your account secure. It’s good 
+>  practice to audit your secure shell keys every so often. Especially if you are using multiple 
+>  computers to access your account.
+{: .callout}
+
+We will run the list command to check what key pairs already exist on your computer.
+
+~~~
+ls -al ~/.ssh
+~~~
+{: .language-bash}
+
+Your output is going to look a little different depending on whether or not SSH has ever been set up on the computer you are using. 
+
+Dracula has not set up SSH on his computer, so his output is 
+
+~~~
+ls: cannot access '/c/Users/Vlad Dracula/.ssh': No such file or directory
+~~~
+{: .output}
+
+If SSH has been set up on the computer you're using, the public and private key pairs will be listed. The file names are either `id_ed25519`/`id_ed25519.pub` or `id_rsa`/`id_rsa.pub` depending on how the key pairs were set up.  
+
+Since they don’t exist on Dracula’s computer, he uses this command to create them: 
+
+~~~
+$ ssh-keygen -t ed25519 -C "vlad@tran.sylvan.ia"
```

Comment:
I know that ssh/ssh-keygen are installed as part of Git Bash from Git for Windows. Are we confident that ssh/ssh-keygen are also available under macOS and Linux, or do we need to amend the lesson setup instructions?

Reply:
Yes, we are confident.